File: backend/text_input.py
import eel
import time

user_input_result = None
waiting_for_input = False

@eel.expose
def submit_text_input(text):
    global user_input_result, waiting_for_input
    user_input_result = text
    waiting_for_input = False
    print(f"✅ Received input: {text[:3]}***")
    return True

def get_text_input(prompt_message, is_password=False):
    global user_input_result, waiting_for_input
    
    try:
        user_input_result = None
        waiting_for_input = True
        
        eel.showTextInputPrompt(prompt_message, is_password)
        
        timeout = 60
        elapsed = 0
        
        while waiting_for_input and elapsed < timeout:
            time.sleep(0.1)
            elapsed += 0.1
        
        if user_input_result is None:
            print("⚠️ Text input timeout")
            return None
        
        result = user_input_result
        user_input_result = None
        waiting_for_input = False
        
        return result
        
    except Exception as e:
        print(f"Error in get_text_input: {e}")
        return None

File: frontend/text_input.js
function showTextInputPrompt(promptMessage, isPassword) {
    console.log("📝 Showing text input prompt:", promptMessage);
    
    $("#promptMessage").text(promptMessage);
    
    const inputField = $("#textInputField");
    inputField.val("");
    inputField.attr("type", isPassword ? "password" : "text");
    inputField.attr("placeholder", isPassword ? "Enter password..." : "Type here...");
    
    $("#textInputModal").fadeIn(300);
    
    setTimeout(() => {
        inputField.focus();
    }, 350);
    
    inputField.off('keypress').on('keypress', function(e) {
        if (e.which === 13) {
            submitTextInput();
        }
    });
}

function submitTextInput() {
    const inputValue = $("#textInputField").val().trim();
    
    if (!inputValue) {
        alert("Please enter a value");
        return;
    }
    
    console.log("✅ Submitting input");
    
    try {
        eel.submit_text_input(inputValue);
    } catch (error) {
        console.error("Error submitting input:", error);
    }
    
    $("#textInputField").val("");
    $("#textInputModal").fadeOut(300);
}

function cancelTextInput() {
    console.log("❌ Input cancelled");
    
    try {
        eel.submit_text_input(null);
    } catch (error) {
        console.error("Error cancelling input:", error);
    }
    
    $("#textInputField").val("");
    $("#textInputModal").fadeOut(300);
}

eel.expose(showTextInputPrompt);